### ISSUE TITLE
Property Links

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#634]
**Describe the project you are working on:**
SpaceShip game

**Describe the problem or limitation you are having in your project:**
I would like to link the root node's script properties to its children node's properties

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
I suggest that property links be added.
basically it would be a editor feature that allows you to link a node's properties to another node's property in the inspector.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
basically there would be something like the keyframe icon on all properties in which you can click and select a node and a property you want to link. 
note the property has to have the same type.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
it can be worked around with a script but its tedious and would be much better in the editor.
 
**Is there a reason why this should be core and not an add-on in the asset library?:**
it helps people make games faster and allows for encapsulation.

**Comment by Kequc** [comment]
The explicit way to do something like this would be a method with a descriptive name like `resize_window(x, y)` which would perform all the necessary operations. I know you describe this as being overly tedious. But what if later you want to be able to modify a property that you've tied up in this way?

Then you' need an additional workaround added to the language.

**Comment by Xrayez** [comment]
I don't quite understand the proposal fully. Does linking properties mean that they are *duplicated* or *synchronized* between node instances? If former, perhaps that has something to do with the trait system as described in godotengine/godot#23101?

**Comment by bojidar-bg** [comment]
Related to https://github.com/godotengine/godot/issues/22263 and https://github.com/godotengine/godot/issues/19073 as well.

**Comment by KoBeWi** [comment]
> it can be worked around with a script but its tedious and would be much better in the editor.

If you use a setter/getter that accesses child property, it's just 5 lines (or 2 if you do one-line functions).

**Comment by Shadowblitz16** [comment]
@Xrayez its synchronized between node instances 

@KoBeWi not if you want to sync variable to multiple nodes
